kite-data/kite-data-core/src/main/java/org/kitesdk/data/Flushable.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2013 Cloudera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kitesdk.data;
+
+/**
+ * A writer that can guarantee data is persisted to disk.
+ * <p>
+ * {@link #flush} a stronger guarantee than {@link java.io.Flushable#flush}.
+ * When it returns, data written is persisted to disk, not just an underlying
+ * buffer.
+ */
+@edu.umd.cs.findbugs.annotations.SuppressWarnings(
+    value="NM_SAME_SIMPLE_NAME_AS_INTERFACE",
+    justification="Intended to be a stricter version of the parent interface")
+public interface Flushable extends java.io.Flushable {
+  /**
+   * Ensure that data has been written to disk.
+   * <p>
+   * This method has a stronger guarantee than {@link java.io.Flushable#flush}.
+   * When this method returns, data written is persisted to disk, not just an
+   * underlying buffer.
+   */
+  @Override
+  void flush();
```

Comment:
> the confusion of two methods, one of which seems reliable but is really not

Reliable is a relative term. Flush (hflush in Hadoop) is reliable in the face of losing power to a rack, but not to a whole data center. Sync (hsync in Hadoop) is reliable in the face of power failure of a whole DC, but not to a DC fire (say). Also, sync is more expensive than flush, and an application may choose flush over sync to trade some reliability for increased throughput. Etc.

So I think we need to expose both flush and sync at the API level in Kite. This can be done through two interfaces, java.io.Flushable and org.kitesdk.data.Syncable:

```java
public interface Syncable {
  void sync() throws IOException;
}
```

Also, we should try out the replacement interfaces to check that they work, pass tests etc before committing this.